src/dune_rules/module_trie.ml

```diff
@@ -49,7 +49,7 @@ let rec gen_set_k t ps v ~on_exists =
         | Some c -> on_exists c)
       else (
         match x with
-        | None -> None
+        | None -> Some (Map (gen_set_k Map.empty ps v ~on_exists))
```

Comment:
Is this code path in use? How about just making it an error if it's not.

Reply:
Do you need this behavior in every code path? If not, a parameter that explicitly allows for overriding would be preferable.

Reply:
One reliable source of bugs is the stdlib's `Map.set` function which freely overrides whatever binding is already present in the map. In stdune, we've decided that a better map api allows us to opt-in into overriding explicitly. You can see this in action with our `Map.add_exn` function which has personally prevented me from introducing many bugs by accident. I think it would be nice to stay consistent with that API for tries, and ask callers to opt-in to overriding bindings.

Reply:
I'm still slightly confused. Doesn't this case seem completely different? We're not overriding here, the bug fix makes it possible to add a missing branch to the `Module_trie`. Am I missing something?

Reply:
Oh yeah, never mind. This is the non case.